build.gradle

```diff
@@ -348,79 +342,101 @@ subprojects {
         from sourceSets.main.allSource
     }
 
-    artifacts { archives javadocJar, sourcesJar }
-
-    uploadArchives.repositories.mavenDeployer {
-        beforeDeployment { MavenDeployment deployment -> signing.signPom(deployment) }
-        if (rootProject.hasProperty('repositoryDir')) {
-            repository(url: new File(rootProject.repositoryDir).toURI())
-        } else {
-            String stagingUrl
-            if (rootProject.hasProperty('repositoryId')) {
-                stagingUrl = 'https://oss.sonatype.org/service/local/staging/deployByRepositoryId/' +
-                        rootProject.repositoryId
-            } else {
-                stagingUrl = 'https://oss.sonatype.org/service/local/staging/deploy/maven2/'
-            }
-            def configureAuth = {
-                if (rootProject.hasProperty('ossrhUsername') && rootProject.hasProperty('ossrhPassword')) {
-                    authentication(userName: rootProject.ossrhUsername, password: rootProject.ossrhPassword)
-                }
-            }
-            repository(url: stagingUrl, configureAuth)
-            snapshotRepository(url: 'https://oss.sonatype.org/content/repositories/snapshots/', configureAuth)
-        }
-    }
-    uploadArchives.onlyIf { !name.contains("grpc-gae-interop-testing") && !name.contains("grpc-xds") }
+    publishing {
+        publications {
+            mavenJava(MavenPublication) {
+                from components.java
 
-    [
-        install.repositories.mavenInstaller,
-        uploadArchives.repositories.mavenDeployer,
-    ]*.pom*.whenConfigured { pom ->
-        pom.project {
-            name "$project.group:$project.name"
-            description project.description
-            url 'https://github.com/grpc/grpc-java'
-
-            scm {
-                connection 'scm:git:https://github.com/grpc/grpc-java.git'
-                developerConnection 'scm:git:git@github.com:grpc/grpc-java.git'
-                url 'https://github.com/grpc/grpc-java'
-            }
+                artifact javadocJar
+                artifact sourcesJar
 
-            licenses {
-                license {
-                    name 'Apache 2.0'
-                    url 'https://opensource.org/licenses/Apache-2.0'
-                }
-            }
+                pom {
+                    name = "$project.group:$project.name".toString()
```

Comment:
Why is toString() necessary?

Reply:
GString(Impl)? seems to have issue (there are few open issues and some are closed but still doesn't quite work)
manual string concat works without toString.